pkg/exporter/sumologicexporter/sender.go

```diff
@@ -57,6 +58,68 @@ type logPair struct {
 	log        pdata.LogRecord
 }
 
+// countingReader keeps number of records related to reader
+type countingReader struct {
+	counter int64
+	reader  io.Reader
+}
+
+// newCountingReader creates countingReader with given number of records
+func newCountingReader(records int) *countingReader {
+	return &countingReader{
+		counter: int64(records),
+	}
+}
+
+func (c *countingReader) withBytes(data []byte) *countingReader {
+	c.reader = bytes.NewReader(data)
+	return c
+}
+
+func (c *countingReader) withString(data string) *countingReader {
+	c.reader = strings.NewReader(data)
+	return c
+}
```

Comment:
* I have a feeling that this whole section is a bit underdocumented.
* Additionally `countingReader` doesn't really count anything, does it? It just holds stuff

Reply:
> Additionally countingReader doesn't really count anything, does it? It just holds stuff

would `readerState` be better choice